app/mbm/static/js/app.js

```diff
@@ -0,0 +1,345 @@
+import UserLocations from './userlocations.js'
+import autocomplete from './autocomplete.js'
+import Geolocation from './geolocation.js'
+// The App class holds top level state and map related methods that other modules
+// need to call, for example to update the position of markers.
+export default class App {
+  constructor(routeListUrl, routeUrl) {
+    this.routeListUrl = routeListUrl
+    this.routeUrl = routeUrl
+
+    this.routeLayer = null
+    this.allRoutesLayer = null
+    this.markers = { 'source': null, 'target': null }
+
+    // Start the app once the DOM is ready
+    document.addEventListener('DOMContentLoaded', this.start.bind(this))
+  }
+
+  start() {
+    // Create the leaflet map
+    this.map = this.createMap()
+
+    // Store references to DOM elements we'll need
+    const $directionsForm = $('#input-elements')
```

Comment:
There was kind of a mixture of jQuery DOM methods and their plain JS equivalents. I didn't make much of an effort to normalize that, but I did prefix jquery vars with `$`